### ISSUE TITLE
cypress thinks clipped elements in overflow-y:hidden elements are visible

### ISSUE THREAD

**Issue by mo** [cypress-io/cypress#563]
Cypress allows `.should('be.visible')` to pass OK on the following:

```html
<div style="display: block; height: 10px; overflow-y: hidden;">
  <br>
  <br>
  <div id="hidden3">hidden3 pass</div>
</div>
```

This page:
https://docs.cypress.io/guides/core-concepts/interacting-with-elements.html#Visibility
...lists "overflow: hidden" but maybe you didn't list the overflow-y property?

**Comment by jennifer-shehane** [comment]
Can confirm, this is still a bug in version 0.20.0. Thanks for opening the issue

**Test code as is in version 0.20.0**
```js
cy.get('#hidden3').should('be.visible')       // PASSES incorrectly
cy.get('#hidden3').should('not.be.visible') // FAILS incorrectly
```

**Comment by chrisbreiding** [comment]
The code for this is done, but this has yet to be released. We'll update the issue and reference the changelog when it's released.

**Comment by brian-mann** [comment]
Fixed in [`0.20.0`](https://docs.cypress.io/guides/references/changelog.html#0-20-0)
